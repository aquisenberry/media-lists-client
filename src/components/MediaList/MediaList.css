.empty {
  width: 100%;
  margin-top: 3rem;

  display: flex;
  justify-content: center;
  align-items: center;
}
.media-list {
  padding-left: 2rem;
}

.media-list__container {
  /* width: 100%; */
  margin-top: 1rem;
  display: flex;
  /* justify-content: center; */
  /* align-items: center; */
  overflow-x: scroll;
  overflow-y: hidden;
  flex-wrap: nowrap;
  flex-direction: row;
  margin-bottom: 1rem;
  padding-left: 0.5rem;
}

@media screen and (max-width: 400px) {
  h1 {
    font-size: 2rem;
  }

  .container {
    margin-top: 2rem;
  }

  .movie {
    width: "100%";
    margin: 1rem;
  }
}
