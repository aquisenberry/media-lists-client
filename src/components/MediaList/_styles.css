.empty {
  width: 100%;
  margin-top: 3rem;

  display: flex;
  justify-content: center;
  align-items: center;
}
.media-list {
  padding-left: 2rem;
}

.media-list__container {
  /* width: 100%; */
  margin-top: 1rem;
  display: flex;
  /* justify-content: center; */
  /* align-items: center; */
  overflow-x: scroll;
  overflow-y: hidden;
  flex-wrap: nowrap;
  flex-direction: row;
  margin-bottom: 1rem;
  padding-left: 0.5rem;
}
/* Modifiers */
.media-list__container--tile{
  width: 100%;
  margin-top: 1rem;
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  align-items: center;
  flex-direction: row;
  margin-bottom: 1rem;
  padding-left: 0.5rem;
}

@media screen and (max-width: 400px) {
  h1 {
    font-size: 2rem;
  }
}
