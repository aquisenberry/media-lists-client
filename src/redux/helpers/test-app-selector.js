
const state = {
    external: {
        searchTerm: "",
        movies:[],
        shows:[],
        videoGames: [],
        boardGames: [],
        books: [],
        mediaDetails: {}
    },
    page: {
        media: [],
        contextMenuAnchor: {
            x:0,
            y:0
        },
        showContextMenu: false,
        contextTarget:{},
        contextType: false
    }
}

export const testUseAppSelector = (f) => f(state)