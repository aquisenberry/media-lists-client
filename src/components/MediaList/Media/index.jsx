/*
    TODO: implement poster minimum height
*/

import React, {useEffect} from 'react'
import  { addMedia, enterMedia, exitMedia } from '../../../redux/actions/page'
import { useAppDispatch } from '../../../redux/helpers/redux-hooks'
import './_styles.css'

const Media = ({media}) => {
    const dispatch = useAppDispatch()
    useEffect(() => {
        dispatch(addMedia({...media, selected: false}))
    }, [dispatch])

    const handleMouseEnterMedia = (e) => {
        dispatch(enterMedia(media))
    }
    const handleMouseExitMedia = (e) => {
        dispatch(exitMedia())
    }
    return (
        <div className="media" onMouseEnter={handleMouseEnterMedia} onMouseLeave={handleMouseExitMedia}>
            <div className="media__image-container">
                <img 
                    className="media__image"
                    src={media.poster}
                    alt={`${media.title} Poster`}
                />
            </div>
            <div className="media-meta">
                <h3 className="media-meta__title">{media.title}</h3>
                <h4 className="media-meta__year">{media.year}</h4>
            </div>
        </div>
    )

}

export default Media