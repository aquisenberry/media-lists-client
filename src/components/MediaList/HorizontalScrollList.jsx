/*
// TODO: implement default image for when images are unavailable within a given list
//       image should be passed as a prop, when the MediaList is called
*/

import React from 'react'
import Media from './Media'

const HorizontalScroll = ({title,mediaList}) => {
    
        return (
            <div className="media-list">
            <h2 className="media-list__title">{title}</h2>
            <div className="media-list__container" >
                {mediaList.map((media) => <Media media={media}/>)}
            </div>
            </div>
        )
}


export default HorizontalScroll