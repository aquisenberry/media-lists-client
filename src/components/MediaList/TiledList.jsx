/*
// TODO: implement default image for when images are unavailable within a given list
//       image should be passed as a prop, when the MediaList is called
*/

import React from 'react'
import Media from './Media'

const TiledList = ({title,mediaList}) => {
    
        return (
            <div className="media-list--tile">
            <h2 className="media-list__title">{title}</h2>
            <div className="media-list__container--tile" >
                {mediaList.map((media) => <Media key={`${media.type}-${media._id}`} media={media}/>)}
            </div>
            </div>
        )
}


export default TiledList