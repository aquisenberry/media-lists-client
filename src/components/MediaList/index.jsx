import React from 'react'
import Media from './Media/altMedia'

import './MediaList.css'
const MediaList = ({mediaList}) => {
    return (
        <div className="media-container" >
            {mediaList.map((media) => <Media media={media}/>)}
        </div>
    )
}

export default MediaList