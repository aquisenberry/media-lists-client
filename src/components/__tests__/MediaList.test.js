import renderer from 'react-test-renderer'
import MediaList from '../MediaList'
import {useAppDispatch, useAppSelector} from '../../redux/helpers/redux-hooks'
import { testUseAppSelector } from '../../redux/helpers/test-app-selector'
import mocks from './__mocks__/MediaList.mocks.json'


jest.mock('../../redux/helpers/redux-hooks', () =>({
    useAppDispatch: jest.fn(() => {}),
    useAppSelector: jest.fn()
}))

describe('MediaList', () =>{
    beforeEach(() =>{
        const dispatch = jest.fn();
        useAppDispatch.mockReturnValue(dispatch);
        useAppSelector.mockImplementation(testUseAppSelector)
    })
    afterEach(() =>{
        jest.clearAllMocks()
    })
    describe('horizontal scroll variant', () => {
        it('has no markup regressions', () =>{
            const component = renderer.create(
                <MediaList title="Movies" mediaList={mocks.movies} />
            )
            const tree = component.toJSON()
            expect(tree).toMatchSnapshot()
        })
        it('displays given title', () =>{
            const component = renderer.create(
                <MediaList title="Shows" mediaList={mocks.shows} />
            )
            const instance = component.root
            expect(instance.findByProps({className: 'media-list__title'}).children).toContain('Shows')
        })
        it('displays the given number of media', () =>{
            const component = renderer.create(
                <MediaList title="Shows" mediaList={mocks.shows} />
            )
            const instance = component.root
            expect(instance.findByProps({className: 'media-list__container'}).children.length).toEqual(6)
        })
        it('is not a tiled container', () =>{
            const component = renderer.create(
                <MediaList title="Shows" mediaList={mocks.shows} />
            )
            const instance = component.root
            expect(instance.findByProps({className: 'media-list'}).children.find((i) => i.type === 'div').props.className).not.toEqual('media-list__container--tiled')
            // expect(instance.findByProps({className: 'media-list__container'}).children.length).toEqual(6)
        })

    })
    describe('tiled variant', () => {
        it('has no markup regressions', () =>{
            const component = renderer.create(
                <MediaList title="Movies" mediaList={mocks.movies} variant="tiled" />
            )
            const tree = component.toJSON()
            expect(tree).toMatchSnapshot()
        })
    })
})