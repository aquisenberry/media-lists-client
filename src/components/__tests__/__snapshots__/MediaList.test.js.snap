// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MediaList horizontal scroll variant has no markup regressions 1`] = `
<div
  className="media-list"
>
  <h2
    className="media-list__title"
  >
    Movies
  </h2>
  <div
    className="media-list__container"
  >
    <div
      className="media"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="media__image-container"
      >
        <img
          alt="Orphan: First Kill Poster"
          className="media__image"
          src="http://image.tmdb.org/t/p/w185/wSqAXL1EHVJ3MOnJzMhUngc8gFs.jpg"
        />
      </div>
      <div
        className="media-meta"
      >
        <h3
          className="media-meta__title"
        >
          Orphan: First Kill
        </h3>
        <h4
          className="media-meta__year"
        >
          2022
        </h4>
      </div>
    </div>
    <div
      className="media"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="media__image-container"
      >
        <img
          alt="Fall Poster"
          className="media__image"
          src="http://image.tmdb.org/t/p/w185/spCAxD99U1A6jsiePFoqdEcY0dG.jpg"
        />
      </div>
      <div
        className="media-meta"
      >
        <h3
          className="media-meta__title"
        >
          Fall
        </h3>
        <h4
          className="media-meta__year"
        >
          2022
        </h4>
      </div>
    </div>
    <div
      className="media"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="media__image-container"
      >
        <img
          alt="Pinocchio Poster"
          className="media__image"
          src="http://image.tmdb.org/t/p/w185/g8sclIV4gj1TZqUpnL82hKOTK3B.jpg"
        />
      </div>
      <div
        className="media-meta"
      >
        <h3
          className="media-meta__title"
        >
          Pinocchio
        </h3>
        <h4
          className="media-meta__year"
        >
          2022
        </h4>
      </div>
    </div>
    <div
      className="media"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="media__image-container"
      >
        <img
          alt="Beast Poster"
          className="media__image"
          src="http://image.tmdb.org/t/p/w185/iRV0IB5xQeOymuGGUBarTecQVAl.jpg"
        />
      </div>
      <div
        className="media-meta"
      >
        <h3
          className="media-meta__title"
        >
          Beast
        </h3>
        <h4
          className="media-meta__year"
        >
          2022
        </h4>
      </div>
    </div>
    <div
      className="media"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="media__image-container"
      >
        <img
          alt="Samaritan Poster"
          className="media__image"
          src="http://image.tmdb.org/t/p/w185/vwq5iboxYoaSpOmEQrhq9tHicq7.jpg"
        />
      </div>
      <div
        className="media-meta"
      >
        <h3
          className="media-meta__title"
        >
          Samaritan
        </h3>
        <h4
          className="media-meta__year"
        >
          2022
        </h4>
      </div>
    </div>
    <div
      className="media"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="media__image-container"
      >
        <img
          alt="RRR Poster"
          className="media__image"
          src="http://image.tmdb.org/t/p/w185/wE0I6efAW4cDDmZQWtwZMOW44EJ.jpg"
        />
      </div>
      <div
        className="media-meta"
      >
        <h3
          className="media-meta__title"
        >
          RRR
        </h3>
        <h4
          className="media-meta__year"
        >
          2022
        </h4>
      </div>
    </div>
    <div
      className="media"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="media__image-container"
      >
        <img
          alt="Thor: Love and Thunder Poster"
          className="media__image"
          src="http://image.tmdb.org/t/p/w185/pIkRyD18kl4FhoCNQuWxWu5cBLM.jpg"
        />
      </div>
      <div
        className="media-meta"
      >
        <h3
          className="media-meta__title"
        >
          Thor: Love and Thunder
        </h3>
        <h4
          className="media-meta__year"
        >
          2022
        </h4>
      </div>
    </div>
    <div
      className="media"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="media__image-container"
      >
        <img
          alt="Prey Poster"
          className="media__image"
          src="http://image.tmdb.org/t/p/w185/ujr5pztc1oitbe7ViMUOilFaJ7s.jpg"
        />
      </div>
      <div
        className="media-meta"
      >
        <h3
          className="media-meta__title"
        >
          Prey
        </h3>
        <h4
          className="media-meta__year"
        >
          2022
        </h4>
      </div>
    </div>
    <div
      className="media"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="media__image-container"
      >
        <img
          alt="Dragon Ball Super: Super Hero Poster"
          className="media__image"
          src="http://image.tmdb.org/t/p/w185/rugyJdeoJm7cSJL1q4jBpTNbxyU.jpg"
        />
      </div>
      <div
        className="media-meta"
      >
        <h3
          className="media-meta__title"
        >
          Dragon Ball Super: Super Hero
        </h3>
        <h4
          className="media-meta__year"
        >
          2022
        </h4>
      </div>
    </div>
    <div
      className="media"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="media__image-container"
      >
        <img
          alt="Fullmetal Alchemist: The Revenge of Scar Poster"
          className="media__image"
          src="http://image.tmdb.org/t/p/w185/c5OwwBNyJkwyroIOIJL9IiRjydR.jpg"
        />
      </div>
      <div
        className="media-meta"
      >
        <h3
          className="media-meta__title"
        >
          Fullmetal Alchemist: The Revenge of Scar
        </h3>
        <h4
          className="media-meta__year"
        >
          2022
        </h4>
      </div>
    </div>
  </div>
</div>
`;

exports[`MediaList tiled variant has no markup regressions 1`] = `
<div
  className="media-list"
>
  <h2
    className="media-list__title"
  >
    Movies
  </h2>
  <div
    className="media-list__container"
  >
    <div
      className="media"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="media__image-container"
      >
        <img
          alt="Orphan: First Kill Poster"
          className="media__image"
          src="http://image.tmdb.org/t/p/w185/wSqAXL1EHVJ3MOnJzMhUngc8gFs.jpg"
        />
      </div>
      <div
        className="media-meta"
      >
        <h3
          className="media-meta__title"
        >
          Orphan: First Kill
        </h3>
        <h4
          className="media-meta__year"
        >
          2022
        </h4>
      </div>
    </div>
    <div
      className="media"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="media__image-container"
      >
        <img
          alt="Fall Poster"
          className="media__image"
          src="http://image.tmdb.org/t/p/w185/spCAxD99U1A6jsiePFoqdEcY0dG.jpg"
        />
      </div>
      <div
        className="media-meta"
      >
        <h3
          className="media-meta__title"
        >
          Fall
        </h3>
        <h4
          className="media-meta__year"
        >
          2022
        </h4>
      </div>
    </div>
    <div
      className="media"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="media__image-container"
      >
        <img
          alt="Pinocchio Poster"
          className="media__image"
          src="http://image.tmdb.org/t/p/w185/g8sclIV4gj1TZqUpnL82hKOTK3B.jpg"
        />
      </div>
      <div
        className="media-meta"
      >
        <h3
          className="media-meta__title"
        >
          Pinocchio
        </h3>
        <h4
          className="media-meta__year"
        >
          2022
        </h4>
      </div>
    </div>
    <div
      className="media"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="media__image-container"
      >
        <img
          alt="Beast Poster"
          className="media__image"
          src="http://image.tmdb.org/t/p/w185/iRV0IB5xQeOymuGGUBarTecQVAl.jpg"
        />
      </div>
      <div
        className="media-meta"
      >
        <h3
          className="media-meta__title"
        >
          Beast
        </h3>
        <h4
          className="media-meta__year"
        >
          2022
        </h4>
      </div>
    </div>
    <div
      className="media"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="media__image-container"
      >
        <img
          alt="Samaritan Poster"
          className="media__image"
          src="http://image.tmdb.org/t/p/w185/vwq5iboxYoaSpOmEQrhq9tHicq7.jpg"
        />
      </div>
      <div
        className="media-meta"
      >
        <h3
          className="media-meta__title"
        >
          Samaritan
        </h3>
        <h4
          className="media-meta__year"
        >
          2022
        </h4>
      </div>
    </div>
    <div
      className="media"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="media__image-container"
      >
        <img
          alt="RRR Poster"
          className="media__image"
          src="http://image.tmdb.org/t/p/w185/wE0I6efAW4cDDmZQWtwZMOW44EJ.jpg"
        />
      </div>
      <div
        className="media-meta"
      >
        <h3
          className="media-meta__title"
        >
          RRR
        </h3>
        <h4
          className="media-meta__year"
        >
          2022
        </h4>
      </div>
    </div>
    <div
      className="media"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="media__image-container"
      >
        <img
          alt="Thor: Love and Thunder Poster"
          className="media__image"
          src="http://image.tmdb.org/t/p/w185/pIkRyD18kl4FhoCNQuWxWu5cBLM.jpg"
        />
      </div>
      <div
        className="media-meta"
      >
        <h3
          className="media-meta__title"
        >
          Thor: Love and Thunder
        </h3>
        <h4
          className="media-meta__year"
        >
          2022
        </h4>
      </div>
    </div>
    <div
      className="media"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="media__image-container"
      >
        <img
          alt="Prey Poster"
          className="media__image"
          src="http://image.tmdb.org/t/p/w185/ujr5pztc1oitbe7ViMUOilFaJ7s.jpg"
        />
      </div>
      <div
        className="media-meta"
      >
        <h3
          className="media-meta__title"
        >
          Prey
        </h3>
        <h4
          className="media-meta__year"
        >
          2022
        </h4>
      </div>
    </div>
    <div
      className="media"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="media__image-container"
      >
        <img
          alt="Dragon Ball Super: Super Hero Poster"
          className="media__image"
          src="http://image.tmdb.org/t/p/w185/rugyJdeoJm7cSJL1q4jBpTNbxyU.jpg"
        />
      </div>
      <div
        className="media-meta"
      >
        <h3
          className="media-meta__title"
        >
          Dragon Ball Super: Super Hero
        </h3>
        <h4
          className="media-meta__year"
        >
          2022
        </h4>
      </div>
    </div>
    <div
      className="media"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="media__image-container"
      >
        <img
          alt="Fullmetal Alchemist: The Revenge of Scar Poster"
          className="media__image"
          src="http://image.tmdb.org/t/p/w185/c5OwwBNyJkwyroIOIJL9IiRjydR.jpg"
        />
      </div>
      <div
        className="media-meta"
      >
        <h3
          className="media-meta__title"
        >
          Fullmetal Alchemist: The Revenge of Scar
        </h3>
        <h4
          className="media-meta__year"
        >
          2022
        </h4>
      </div>
    </div>
  </div>
</div>
`;
